test/recipes/80-test_ssl_old.t

```diff
@@ -822,4 +822,37 @@ sub testssl {
         ok(run(test([@ssltest, "-bio_pair", "-tls1", "-ct"])));
         }
     };
+
+    subtest 'Session ticket + SNI enable tests' => sub {
+	######################################################################
```

Comment:
I.e. what we need to do here:

1) Extend the tests with a "server2" configuration section. Add a second server CTX to handshake_helper.

2) Add the following options to the [test] section:

ServerSessionTicketKeyCallback - Default (built-in) / NoTicket ("broken"). Only has effect when session tickets are enabled. You can also add a Server2SessionTicketKeyCallback, though it should be sufficient to only ever explicitly configure this on the first CTX? The third, crashing option can be replaced by recording & verifying that the callback was called on the right ctx.

ServerName - string value; the servername the client should connect to.

For SNI, I think it's fine for now to bake in two names into handshake_helper (I'd suggest "server" and "server2" to match the conf sections). Then, just configure a SNI switching callback to always switch between the two (and simply do nothing whenever the client didn't send a servername).

It should be mostly straightforward copy-paste from existing code. But it's quit some changes so let me know if you'd like me to do parts of it!